src/Commands/GenerateModule.php

```diff
@@ -0,0 +1,704 @@
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Exception\IOExceptionInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+
+
+
+/**
+ * This command generate a module with all their basic files and directories.
+ */
+final class GenerateModule extends Command
+{
+    private $tabHook=array();
+    private $tabSQL=array();
+	private $tabTabs=array();
+    private $hookStringInstall="";
+    private $hookStringFunction="";
+    private $sqlTableStringInstall="";
+    private $sqlTableStringUninstall="";
+    private $sqlTableStringFunction="";
+	private $tabStringInstall="";
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('fop:module:generate')
+            ->setDescription('create a new module')
+            ->setHelp('This command instantiate every files and directores of Symfony/Prestashop in Console Context')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $filesystem = new Filesystem();
+
+        $io->title('This program was carried out by FriendsOfPresta');
+        $io->warning('Be careful if you run this program it may erase data, be sure to specify an unused module name or you know what you are doing');
+
+        $helper = $this->getHelper('question');
+        $question = new ChoiceQuestion(
+            'Would like to continue (defaults to No)',
+            ['No', 'Yes'],
+            0
+        );
+        $question->setErrorMessage('Incorrect number, enter 0 for No and 1 for Yes.');
+        $confirm = $helper->ask($input, $output, $question);
+        if ($confirm == "No") {
+            $io->text("End of the program ...");
+            return;
+        }
+
+        $io->newLine(1);
+        $io->section('Edition of the entry point');
+
+        $name = $io->ask('Give a name to your module', 'fop_examplemodule');
+        $displayName = $io->ask('Give a display name to your module', 'FOP Exemple Module');
+        $fileName = strtolower($name);
+        $filenameFirstletterCaps = ucfirst($name);
+        $filenameCamel = ucwords(strtolower($name));
+        $description = $io->ask("Give the description of the module");
+        $author = $io->ask("Give the author of the module", 'FOP');
+        $mailContact = $io->ask("Give your contact email to reach you in case of problems","Generic@email.com");
+        
+        $io->section('File generation');
+
+        $fileDirectories = array();
+        try {
+            $fileDirectories["dir"][] = _PS_MODULE_DIR_."".$fileName;
```

Comment:
I'm wondering if it necessary to create all the files and directory ?
With a simple case of module you've got a lot of directories